Assets/Scripts/StandardSamples/UI/Common/SampleMenu.cs

```diff
@@ -0,0 +1,316 @@
+        /// </summary>
+        public virtual void Show()
+        {
+            Log($"Show() started");
+
+            // Don't do anything if already showing.
+            if (!Hidden) return;
+
+            if (null == UIParent)
+            {
+                Log($"UIParent for is null for some reason.");
+            }
+            else
+            {
+                UIParent.SetActive(true);
+            }
+
+            // Make the gameObject active. This should make all the relevant
+            // items in the sample menu visible and interactable. 
+            if (null == gameObject)
+            {
+                Log($"GameObject for is null for some reason.");
+                return;
+            }
+            else
+            {
+                gameObject.SetActive(true);
+            }
+
+            // Set the selected control to be the first selected element, so
+            // long as that element can actually be selected.
+            SetSelected();
+
+            // Flag as showing.
+            Hidden = false;
+
+            Log($"Show() completed");
+        }
+
+        /// <summary>
+        /// Hides the SampleMenu. If overriding, make sure to call this base
+        /// implementation first.
+        /// </summary>
+        public virtual void Hide()
+        {
+            Log($"Hide() started");
+
+            // Don't do anything if already hidden.
+            if (_hidden.HasValue && _hidden.Value) return;
+
+            if (null == UIParent)
+            {
+                Log($"UIParent is null for some reason.");
+            }
+            else
+            {
+                UIParent.SetActive(false);
+            }
+
+            // Make the gameObject inactive. This should make all the relevant 
+            // items in the sample menu invisible and non-interactable.
+            gameObject.SetActive(false);
+
+            // Flag as hidden.
+            Hidden = true;
+
+            Log($"Hide() completed");
+        }
+
+        /// <summary>
+        /// Sets the focused control to UIFirstSelected, assuming it is both
+        /// defined and active in the hierarchy.
+        /// </summary>
+        private void SetSelected()
+        {
+            // Controller: Detect if nothing is selected and controller input detected, and set default
+            if (UIFirstSelected == null 
+                || UIFirstSelected.activeSelf == false
+                || UIFirstSelected.activeInHierarchy == false
+                || EventSystem.current == null || EventSystem.current.currentSelectedGameObject != null
+                || !InputUtility.WasGamepadUsedLastFrame())
+            {
+                return;
+            }
+
+            // Controller
+            EventSystem.current.SetSelectedGameObject(UIFirstSelected);
+            Debug.Log("Nothing currently selected, default to UIFirstSelected: EventSystem.current.currentSelectedGameObject = " + EventSystem.current.currentSelectedGameObject);
```

Comment:
Can you pass the currentSelectedGameObject as a parameter to Debug.Log? It'll make it so that if someone clicks on the log, it'll select that game object in the hierarchy.

Reply:
Either that, and/or it should be using the below Log function.

Reply:
Yep, good catch. Fixed.